strawberryfields/decompositions.py

```diff
@@ -251,7 +254,10 @@ def nullT(n, m, U):
     if nmax != mmax:
         raise ValueError("U must be a square matrix")
 
-    if U[n-1, m] == 0:
+    if (U[n-1, m] == 0 and U[n, m] == 0):
```

Comment:
```suggestion
    if (U[n, m] == 0):
```
same comment as above